tests/integration/utils.sh

```diff
@@ -167,6 +167,106 @@ teardown_container_environment()
   done
 }
 
+# completely remove the container environment: the containers and the images
+function teardown_container_environment()
+{
+  local container_name
+  local distro
+  local img
+  local img_layers
+  local tmp_img
+  local tmp_img_layers
+  local flags
+
+  # flag to force-remove images
+  if [[ "$1" == '-f' ]]; then
+    flags='--force'
+  fi
+
+  for distro in "${DISTROS[@]}"; do
+    container_name="kw-${distro}"
+    podman container exists "${container_name}"
+    if [[ "$?" -eq 0 ]]; then
+      say "Removing podman container kw-${distro}."
+      teardown_single_container "${container_name}"
+    fi
+
+    img="kw-${distro}"
+    podman image exists "${img}"
+    if [[ "$?" -eq 0 ]]; then
+      say "Removing podman ${img} image."
+
+      # get this image layers, so we can safely remove its children dangling images.
+      img_layers=$(image_inspect -f '{{.RootFS.Layers}}' "${img}" | tr -d '[]')
+
+      # remove possible dangling images
+      image_ls --all --quiet --filter 'intermediate=true' | while read -r tmp_img; do
+
+        # we check if the dangling image have layers which are exact prefixes of
+        # the current image we are trying to delete.
+        tmp_img_layers=$(image_inspect -f '{{.RootFS.Layers}}' "${tmp_img}" | tr -d '[]')
+        grep --perl-regexp "^${tmp_img_layers}" <<< "${img_layers}" > /dev/null
+
+        # delete the dangling image if that is the case.
+        if [[ "$?" -eq 0 ]]; then
```

Comment:
A cleaner way to do this would be
```bash
var=$(printf '%s' "$img_layers" | grep --perl-regexp "^${tmp_img_layers}")
if [[ -n "$var" ]]; then
 ...
```
Of course, change the name `var` to a more adequate one.

I say that this is cleaner, because we don't need to redirect the output to `/dev/null`, reading it from left to right (beginning to end) makes more sense, and, most importantly, it follows our coding style.